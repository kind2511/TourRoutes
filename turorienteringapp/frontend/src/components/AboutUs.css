body, html {
    margin: 0;
    padding: 0;
    overflow: hidden; /* Prevent scrolling on the page */
}

.aboutus-container {
    padding: 20px;
    max-width: 800px;
    margin: 0 auto; /* Center the container horizontally */
}

.aboutus-title {
    font-size: 1.5rem;
    text-align: center; /* Center the title */
    font-weight: bold;
}

.aboutus-text {
    font-size: 1rem;
    line-height: 2.9; /* line spacing */
    margin-bottom: 20px; /* Space between the text and the logo */
}

.aboutus-logo {
    font-family: 'Pacifico', cursive;
    font-size: 2rem;
    color: #4CAF50;
    position: absolute;
    bottom: 10px;
    right: 10px;
    cursor: pointer;
    font-weight: bold;
}
