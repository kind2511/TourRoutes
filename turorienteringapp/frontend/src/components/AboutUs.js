import React from 'react';
import { useNavigate } from 'react-router-dom';
import './AboutUs.css';

function AboutUs() {
    // Text for the about us section
    const aboutText = `
        TurRuter has a big map of the whole world. You can see names, places, and addresses everywhere. 
        You can draw lines on the map to make routes from one place to another. Our tool figures out how far it is and 
        keeps the info safe for you. You can keep your routes private if you don't want others to see them. We care about 
        your privacy. Our tool helps you find the quickest way between two places. Just tell it where to go, and it shows 
        you the way. Our app is easy to use and looks nice. It's made for everyone to enjoy using. We keep all your data 
        safe. You can trust us with your routes and information. Our application is more than just maps. It's a tool for 
        exploring the world easily to make your travel easy, efficient, and enjoyable.
    `;

    // Navigate function from react-router-dom
    const navigate = useNavigate();

    // Handle logo click to navigate to the homepage
    const handleLogoClick = () => {
        navigate('/');
    };

    return (
        <div className="aboutus-container">
            <div className="aboutus-title">About Us</div>
            <div className="aboutus-text">{aboutText}</div>
            <div className="aboutus-logo" onClick={handleLogoClick}>TurRuter</div>
        </div>
    );
}

export default AboutUs;
