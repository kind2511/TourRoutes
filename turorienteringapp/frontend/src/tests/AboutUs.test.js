import React from 'react';
import { render } from '@testing-library/react';
import { BrowserRouter } from 'react-router-dom';
import AboutUs from '../components/AboutUs';

test('AboutUs component renders without crashing', () => {
  render(
    <BrowserRouter>
      <AboutUs />
    </BrowserRouter>
  );
});

//----------------------------------------------->

//  Modify Snapshot test for the about us component
test('AboutUs component content verification', () => {
  const { asFragment } = render(
    <BrowserRouter>
      <AboutUs />
    </BrowserRouter>
  );
  expect(asFragment()).toMatchSnapshot();

});
