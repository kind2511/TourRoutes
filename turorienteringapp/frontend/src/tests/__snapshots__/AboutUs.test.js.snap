// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AboutUs component content verification 1`] = `
<DocumentFragment>
  <div
    class="aboutus-container"
  >
    <div
      class="aboutus-title"
    >
      About Us
    </div>
    <div
      class="aboutus-text"
    >
      
        TurRuter has a big map of the whole world. You can see names, places, and addresses everywhere. 
        You can draw lines on the map to make routes from one place to another. Our tool figures out how far it is and 
        keeps the info safe for you. You can keep your routes private if you don't want others to see them. We care about 
        your privacy. Our tool helps you find the quickest way between two places. Just tell it where to go, and it shows 
        you the way. Our app is easy to use and looks nice. It's made for everyone to enjoy using. We keep all your data 
        safe. You can trust us with your routes and information. Our application is more than just maps. It's a tool for 
        exploring the world easily to make your travel easy, efficient, and enjoyable.
    
    </div>
    <div
      class="aboutus-logo"
    >
      TurRuter
    </div>
  </div>
</DocumentFragment>
`;

exports[`AboutUs component snapshot 1`] = `
<DocumentFragment>
  <div
    class="container"
  >
    <div
      class="title"
    >
      About Us
    </div>
    <div
      class="segment"
    >
      TurRuter provides accurate navigation solutions for urban explorers. We integrate the latest technology to ensure you always find your way, whether in familiar terrains or new cities.
    </div>
    <div
      class="segment"
    >
      Our system is designed to personalize your experience. TurRuter records and recalls places you've visited, making it easy to return to spots you enjoyed or to plan new routes based on your preferences.
    </div>
    <div
      class="segment"
    >
      TurRuter offers real-time updates to ensure users are informed about any unexpected changes in their routes. Whether there's a sudden roadblock or an event, our app ensures youâ€™re provided with the most efficient detours.
    </div>
    <div
      class="segment"
    >
      TurRuter encourages a sense of community among its users. Discover something interesting? Share it on the platform. Likewise, benefit from the experiences and discoveries of others.
    </div>
    <div
      class="segment"
    >
      User feedback is a cornerstone of TurRuter's development. We are constantly seeking to improve and update our services based on the needs and preferences of our users.
    </div>
    <div
      class="segment"
    >
      With growing concerns about the environment, TurRuter provides eco-friendly route options. This feature supports users in making sustainable travel choices.
    </div>
    <div
      class="conclusion"
    >
      TurRuter is not just a navigation tool; it's a comprehensive platform aiming to enhance the overall travel experience. Our commitment is to ensure convenience, reliability, and innovation in every journey you undertake.
    </div>
    <div
      class="aboutus-logo"
    >
      TurRuter
    </div>
  </div>
</DocumentFragment>
`;
